browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,472 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_cosmetic_resources_tab_helper.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "base/strings/stringprintf.h"
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "brave/content/browser/cosmetic_filters_communication_impl.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
```

Comment:
in theory this tab helper should rather stay in `src/components/`, most deps on `chrome` look resolvable.  Just an idea for a follow-up

Reply:
Created a follow up https://github.com/brave/brave-browser/issues/13473